command/state_mv.go

```diff
@@ -287,22 +289,17 @@ func (c *StateMvCommand) Run(args []string) int {
 					// If we're moving to an address without an index then that
 					// suggests the user's intent is to establish both the
 					// resource and the instance at the same time (since the
-					// address covers both), but if there's an index in the
-					// target then the resource must already exist.
+					// address covers both). If there's an index in the
+					// target then allow creating the new instance here,
+					// inferring the mode from how the new address was parsed.
 					if addrTo.Resource.Key != addrs.NoKey {
-						diags = diags.Append(tfdiags.Sourceless(
-							tfdiags.Error,
-							msgInvalidTarget,
-							fmt.Sprintf("Cannot move to %s: %s does not exist in the current state.", addrTo, addrTo.ContainingResource()),
-						))
-						c.showDiagnostics(diags)
-						return 1
+						fromEachMode = eachModeForInstanceKey(addrTo.Resource.Key)
 					}
 
 					resourceAddr := addrTo.ContainingResource()
```

Comment:
Do we need to do anything special here for the situation where the target resource address is in a module that doesn't exist in the state yet? I don't recall if `SyncWrapper().SetResourceMeta` handles automatically creating the module shell in that case; if it does, I think it could help to note that specifically in a comment here to aid future maintainers.

Reply:
Nope, moving to a new module already works correctly. I'm going to double check that we have that in a test, or update this one to cover it.